### ISSUE TITLE
Screen reader is not announcing label name for section present under stepper variants in Windows.

### ISSUE THREAD

**Issue by msftedad** [angular/components#32037]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

**User Impact:**
Visually impaired users who rely on screen readers will be impacted if Screen reader is not announcing label name for section present under stepper variants in Windows. They won’t know which section they are in, leading to confusion and difficulty completing tasks.

### Reproduction

URL: https://material.angular.dev/components/stepper/overview

**Steps to reproduce:**
1. Enable NVDA and open the URL: https://material.angular.dev/components/stepper/overview in edge browser.
2. Landing page will get open.
3. Press tab key to reach to the sections present under 'Stepper Variants'.
4. Observe whether screen reader is announcing label name for section present under stepper variants in Windows or not.

### Expected Behavior

Screen reader should announce the label name for section present under stepper variants in Windows.

### Actual Behavior

Screen reader is not announcing label name for section present under stepper variants in Windows.

**Refer Attachment:**
Screen reader is not announcing label name for section present under stepper variants in Windows.png
Screen reader is not announcing label name for section present under stepper variants in Windows.mp4

https://github.com/user-attachments/assets/9e5ee684-4563-444b-8f76-0c69297de5fc
<img width="960" height="578" alt="Image" src="https://github.com/user-attachments/assets/d97148c2-708f-4f1d-a2a3-97f608276dde" />

### Environment

OS: Windows 11 25H2 (OS Build 26200.6584)
Browser: Edge (Version 140.0.3485.94)
Angular 20.x.
Material: 20.2.7
Screen Reader: NVDA (2025.2), JAWS (2025.2508.120), Narrator
URL: https://material.angular.dev/components/stepper/overview

**Comment by crisbeto** [comment]
I'm not sure which section you're referring to. Is it the stepper header sections?

**Comment by msftedad** [comment]
Hi @crisbeto , Thanks for the follow up. We are checking the details with dev team. Will confirm back once we hear from them.

**Comment by adolgachev** [comment]
Closing for now, please re-open if get further clarification as requested on the exact place where there is no or insufficient announcement, as well as a snapshot of the DOM there. If you can also test on other browsers, that would be extremely helpful to find out if this is an Edge specific issue.
